openid/idtokenvalidator.go

```diff
@@ -140,25 +140,23 @@ func validateAudiences(jt *jwt.Token, p *Provider) (string, error) {
 		return "", err
 	}
 
-	for _, aud := range p.ClientIDs {
-		for _, audienceClaim := range audiencesClaim {
-			ta, ok := audienceClaim.(string)
-			if !ok {
-				fmt.Printf("aud type %T \n", audienceClaim)
-				return "", &ValidationError{Code: ValidationErrorInvalidAudienceType, Message: fmt.Sprintf("Invalid Audiences type: %T", audiencesClaim), HTTPStatus: http.StatusUnauthorized}
```

Comment:
For the requirement of skipping the audiences validation altogether we should just allow the user to provide his own implementation of the validateAudiences function. If not provided then we can fall back to the internal validatedAudiences without changes. That extension point should be registered in the middleware configuration as another option.